testing/dart/locale_test.dart

```diff
@@ -19,4 +19,44 @@ void main() {
     expect(const Locale('iw', 'DD').toString(), 'he_DE');
     expect(const Locale('iw', 'DD'), const Locale('he', 'DE'));
   });
+
+  test('Locale.fromSubtags', () {
+    expect(const Locale.fromSubtags().languageCode, 'und');
+    expect(const Locale.fromSubtags().scriptCode, null);
+    expect(const Locale.fromSubtags().countryCode, null);
+    expect(const Locale.fromSubtags().variants, orderedEquals([]));
+
+    expect(const Locale.fromSubtags(language: 'en').toString(), 'en');
+    expect(const Locale.fromSubtags(language: 'en').languageCode, 'en');
+    expect(const Locale.fromSubtags(script: 'Latn').toString(), 'und_Latn');
+    expect(const Locale.fromSubtags(script: 'Latn').scriptCode, 'Latn');
+    expect(const Locale.fromSubtags(region: 'US').toString(), 'und_US');
+    expect(const Locale.fromSubtags(region: 'US').countryCode, 'US');
```

Comment:
The examples above demonstrate an unfortunate mismatch between constructor named parameters and the resulting members (region -> countryCode, script -> scriptCode, language -> languageCode). How uncomfortably unidiomatic is this?

Reply:
I'd say "somewhat unidiomatic", but it's hard to search for. 
More importantly, the expression *is* using two different names about *the same thing*. That's a red usability flag, it makes it hard for users to remember which name goes where. 

I'd use the same name in both places.
I kind-of prefer `countryCode` to `country` since it *is* a code. The string "DK" is not a country, it's a string containing a code representing a country.

It's longer, though. If people are going to write it a lot, and it's somehow given from context that it is a code, then going `language` in both cases can also be defended. I'd ask a usability specialist on that, and base it on actual uses, not me speculating.

Reply:
Other considerations:
- "regionCode" would also be more accurate than "countryCode", it might be a non-country region (es-419 is a common example, "Spanish appropriate for the Latin America and Caribbean region").
- renaming members would be hard: breaks backward compatibility for existing Flutter code.

In that case we may need to be going in this direction:

Locale.fromSubtags(languageCode: 'zh', scriptCode: 'Hans', countryCode: 'CN')

(Much as I wish this would be simply Locale('zh', 'Hans', 'CN') ... but then that again requires conditionals and branches in this constructor, so we're abandoning this idea for now.)